### PR TITLE
task 7 post-api-articles-article_id-comments pull request

### PR DESCRIPTION
**Pull request by Bboymad**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -218,5 +218,55 @@ describe('GET /api/articles', () => {
...
+        expect(body.comment).toMatchObject({
+          comment_id: expect.any(Number),
+          body: expect.any(String),
+          article_id: expect.any(Number),
+          author: expect.any(String),
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+        });
+      })
+    })
```
Comment (mezzDavies) [inline __tests__/app.test.js]: You could definitely combine these two test into one 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -218,5 +218,55 @@ describe('GET /api/articles', () => {
...
+        body: 'This is a test comment!'
+      }
+      return request(app)
+      .post('/api/articles/2/comments')
+      .send(newComment)
+      .then(({body}) => {
+        expect(body.comment).toMatchObject({
+          comment_id: expect.any(Number),
+          body: expect.any(String),
+          article_id: expect.any(Number),
```
Comment (mezzDavies) [inline __tests__/app.test.js]: WE could be more explicit here and assert the correct/matching article id here

File: controllers/controller.aritcles.js [hunk 3]
```diff
@@ -44,3 +43,17 @@ exports.getArticles = (req, res, next) => {
   next(err))
   };
 
+exports.postComment = (req, res, next) => {
+  const newComment = req.body;
+  const { article_id } = req.params;
+  console.log("newComment >", newComment)
```
Comment (mezzDavies) [inline controllers/controller.aritcles.js]: Remove console.logs

File: models/models.articles.js [hunk 4]
```diff
@@ -61,3 +60,18 @@ exports.selectArticles = () => {
...
+
+  return db.query(`
+  INSERT INTO comments
+  (body, article_id, author)
+  VALUES
+  ($1, $2, $3)
+  RETURNING *;
+`, [body, article_id, username])
+  .then(({rows}) => {
+    console.log("rows >>>", rows)
```
Comment (mezzDavies) [inline models/models.articles.js]: 👀

File: controllers/controller.aritcles.js [hunk 5]
```diff
@@ -44,3 +43,17 @@ exports.getArticles = (req, res, next) => {
...
+  console.log({ article_id })
+  insertComment(article_id, newComment)
+  .then((comment) => {
+    console.log("comment >>>", comment)
```
Comment (mezzDavies) [inline controllers/controller.aritcles.js]: 👀

**Review by mezzDavies — COMMENTED** [review]
This is heading in the right direction but there are some errors still to consider:

What if a user doesn't supply the request properly - it might be missing properties for example?

What if the article id used in the url is valid (ie a number) but but doesn't exist eg 99999?

What if the username included in the request is not found in the DB?

You should have error handlers in place so should be a matter of writing the tests and then acting accordingly for PSQL errors and/or making a custom error

File: __tests__/app.test.js (hunk 2 above)
Comment (Harrison365) [reply __tests__/app.test.js]: Still needed ^^^

File: __tests__/app.test.js [hunk 6]
```diff
@@ -218,5 +155,81 @@ describe('GET /api/articles', () => {
...
+      return request(app)
+        .post('/api/articles/1/comments')
+        .send(newComment)
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe('Not found');
+        });
+    });
+  });
+});
```
Comment (Harrison365) [inline __tests__/app.test.js:250]: One more test for if the req object has no body property.

**Review by 2202Hannah — APPROVED** [review]
Hey :) great work implementing the changes based on Mezz and Harrison's feedback. 

You are still not asserting the exact article_id in your test - to clarify instead of `article_id: expect.any(Number)` you can expect `article_id: 2` as we would expect that every comment in the response would be for article 2.

Once you have tweaked this test you can merge to main

**Pull request merged**